deepcave/utils/symbolic_regression.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import sympy
+from gplearn import functions
+from gplearn.functions import make_function
+
+from deepcave.utils.logs import get_logger
+
+logger = get_logger(__name__)
+
+
+# Create a safe exp function which does not cause problems
+def exp(x):
+    with np.errstate(all="ignore"):
+        max_value = np.full(shape=x.shape, fill_value=100000)
+        return np.minimum(np.exp(x), max_value)
+
+
+def get_function_set():
+    exp_func = make_function(function=exp, arity=1, name="exp")
+
+    function_set = ["add", "sub", "mul", "div", "sqrt", "log", "sin", "cos", "abs", exp_func]
+
+    return function_set
+
+
+def convert_symb(symb, n_decimals: int = None, hp_names: list = None) -> sympy.core.expr:
+    """
+    Convert a fitted symbolic regression to a simplified and potentially rounded mathematical expression.
```

Comment:
This line is over 100 characters; however, I am not 100% sure what our desired setting for the linter would be. 100 is admittedly really short.